File: collector/src/main/java/com/bitmovin/analytics/stateMachines/StateMachineListener.java
package com.bitmovin.analytics.stateMachines;

import com.bitmovin.analytics.data.ErrorCode;

public interface StateMachineListener {

    void onStartup(PlayerStateMachine stateMachine, long videoStartupTime, long playerStartupTime);

    void onPauseExit(PlayerStateMachine stateMachine, long duration);

    void onPlayExit(PlayerStateMachine stateMachine, long duration);

    void onHeartbeat(PlayerStateMachine stateMachine, long duration);

    void onRebuffering(PlayerStateMachine stateMachine, long duration);

    void onError(PlayerStateMachine stateMachine, ErrorCode errorCode);

    void onSeekComplete(PlayerStateMachine stateMachine, long duration);

    void onAd(PlayerStateMachine stateMachine);

    void onMute(PlayerStateMachine stateMachine);

    void onUnmute(PlayerStateMachine stateMachine);

    void onUpdateSample(PlayerStateMachine stateMachine);

    void onQualityChange(PlayerStateMachine stateMachine);

    void onVideoChange(PlayerStateMachine stateMachine);

    void onSubtitleChange(PlayerStateMachine stateMachine);

    void onAudioTrackChange(PlayerStateMachine stateMachine);

    void onVideoStartFailed(PlayerStateMachine stateMachine);
}

File: collector/src/main/java/com/bitmovin/analytics/stateMachines/StateMachineListener.kt
package com.bitmovin.analytics.stateMachines

import com.bitmovin.analytics.dtos.ErrorCode
import com.bitmovin.analytics.dtos.SubtitleDto

interface StateMachineListener {
    fun onStartup(
        stateMachine: PlayerStateMachine,
        videoStartupTime: Long,
        playerStartupTime: Long,
    )

    fun onPauseExit(
        stateMachine: PlayerStateMachine,
        duration: Long,
    )

    fun onPlayExit(
        stateMachine: PlayerStateMachine,
        duration: Long,
    )

    fun onTriggerSample(
        stateMachine: PlayerStateMachine,
        duration: Long,
        ssaiRelated: Boolean,
    )

    fun onRebuffering(
        stateMachine: PlayerStateMachine,
        duration: Long,
    )

    fun onError(
        stateMachine: PlayerStateMachine,
        errorCode: ErrorCode?,
    )

    fun onSeekComplete(
        stateMachine: PlayerStateMachine,
        duration: Long,
    )

    fun onAd(
        stateMachine: PlayerStateMachine,
        duration: Long,
    )

    fun onMute(stateMachine: PlayerStateMachine)

    fun onUnmute(stateMachine: PlayerStateMachine)

    fun onUpdateSample(stateMachine: PlayerStateMachine)

    fun onQualityChange(stateMachine: PlayerStateMachine)

    fun onVideoChange(stateMachine: PlayerStateMachine)

    fun onSubtitleChange(
        stateMachine: PlayerStateMachine,
        oldValue: SubtitleDto?,
    )

    fun onAudioTrackChange(stateMachine: PlayerStateMachine)

    fun onVideoStartFailed(
        stateMachine: PlayerStateMachine,
        durationInStartupStateMs: Long,
    )
}
